### ISSUE TITLE
ORT arm cannot run granite-1b (Attention-24): inconsistent total_sequence_length between attn_mask and past_key

### ISSUE THREAD

**Issue by justinchuby** [justinchuby/onnx-genai#1625]
## Summary

The ORT backend **cannot run** `granite-1b-a400m-f16-mobius` at all — on **either** the CUDA EP or the CPU EP. This blocks any native-vs-ORT comparison for this model, and means previously quoted ORT numbers for granite should be treated as unverified.

## Reproduction (measured)

```
profile_native --model <granite-1b-a400m-f16-mobius> --ep cuda --backend ort --steady --tokens 128 --warmups 2 --runs 3
```

```
[E:onnxruntime:, sequential_executor.cc:671] Non-zero status code returned while running Attention node.
Name:'model/layers.0/self_attn/Attention_node_29'
Status Message: onnxruntime/core/providers/cpu/llm/attention_helper.h:147
  ComputeOutputShapeForAttention
  attn_mask == nullptr || attn_mask->Shape()[NumDimensions()-1] == parameters.total_sequence_length was false.
  inconsistent total_sequence_length (between attn_mask and past_key and past_value)
```

Identical failure with `--ep cpu`, so this is **not** a CUDA-provider configuration problem.

## Control (measured)

`qwen05b-fresh` runs fine under the **same** ORT configuration in the same session — 136.29 tok/s on the CUDA EP. So the failure is specific to this model, not to the harness setup or the ORT install.

## Why the two models differ

granite uses the **standard opset-24 `ai.onnx::Attention`**:

```
domain: ''   name: model/layers.0/self_attn/Attention_node_29
attrs: is_causal=1, kv_num_heads=8, q_num_heads=16
inputs: [q, k, v, mask, past_key_values.0.key, past_key_values.0.value]
```

ORT validates that the mask's last dimension equals `total_sequence_length`. GQA-style models that take `seqlens_k` (e.g. `com.microsoft::GroupQueryAttention`) carry no such constraint, which is why qwen05b is unaffected.

**The exported contract itself looks correct** — the graph declares exactly what ORT wants:

```
attention_mask         [batch, 'past_seq_len + seq_len']
past_key_values.N.key  [batch, 8, 'past_sequence_len', 64]
```

So the inconsistency is in **what our ORT arm feeds at runtime**, not in the model's declared shapes: the mask width and the past length handed to ORT disagree. A padded-capacity mask against a logical-length past (or vice versa) would produce exactly this. Note this is the same logical-vs-physical KV length theme as #1491, but on the ORT feeding path rather than the native kernel.

## Impact

- **No native-vs-ORT comparison is currently possible for granite.** Native runs fine and, since #1491, now captures: `cuda_graph: enabled=true captures=5 replays=630 fallbacks=0`, 31.02 tok/s (RTX 4060 Laptop, 128 tokens, 2 warmups, 3 runs, quiet box).
- Any previously quoted ORT throughput for granite (e.g. the `~9.9 tok/s` figure used when native was capture-declined) is **unverified** and should not be reused until this is fixed — the ORT arm cannot execute the model.
- It likely affects **every** model exported with standard `Attention-24` plus a mask, not just granite.

## Suggested next step

Instrument the ORT arm to print the actual fed shapes for `attention_mask` and `past_key_values.0.key` on the failing step, and make them agree on `total_sequence_length`.

Found while measuring granite after the #1491 capture fix.
